docs/decisions/0017-reimplement-asset-processing.rst

```diff
@@ -0,0 +1,268 @@
+     - JS modules and CSS files installed via NPM. Not committed to edx-platform.
+     - React, studio-frontend, paragon
+     - Uninstall as part of full LMS/CMS frontend replatforming
+   * - **XModule Fragments**
+     - JS and SCSS belonging to the older XModule-style XBlocks defined in edx-platform
+     - ProblemBlock (aka CAPA) assets
+     - Convert to pure XBlock fragments
+   * - **XBlock Fragments**
+     - JS and CSS belonging to the pure XBlocks defined in edx-platform
+     - library_sourced_block.js
+     - Keep and/or extract to pip-installed, per-XBlock repositories
+   * - **pip-installed Assets**
+     - Pre-compiled static assets shipped with several Python libraries that we install, including XBlocks. Not committed to edx-platform.
+     - Django Admin, Swagger, Drag-And-Drop XBlock V2
+     - Keep
+
+*Note: this table excludes HTML templates. Templates are part of the frontend, but they are dynamically rendered by the Web application and therefore must be handled differently than static assets.*
+
+So, with the exception of XBlock fragments and pip-installed assets, which are very simple for edx-platform to handle, we plan to eventually remove all edx-platform static frontend assets. However, given the number of remaining edx-platform frontends and speed at which they are currently being replatformed, estimates for completion of this process range from one to five years. Thus, in the medium term future, we feel that timeboxed improvements to how edx-platform handles static assets are worthwhile, especially when they address an acute pain point.
+
+Current pain points
+===================
+
+Three particular issues have surfaced in Developer Experience Working Group discussions recently, each with some mitigations involving static assets:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Pain Point
+     - Potential solution(s)
+
+   * - edx-platform Docker images are too large and/or take too long to build.
+     - Switch from large, legacy tooling packages (such as libsass-python and paver) to industry standard, pre-compiled ones (like node-sass or dart-sass). Remove unnecessary & slow calls to Django management commands.
+
+   * - edx-platform Docker image layers seem to be rebuilt more often than they should.
+     - Remove all Python dependencies from the static asset build process, such that changes to Python code or requirements do not always have to result in a static asset rebuild.
+
+   * - In Tutor, using a local copy of edx-platform overwrites the Docker image's pre-installed node_modules and pre-built static assets, requiring developers to reinstall & rebuild in order to get a working platform.
+     - Better parameterize the input and output paths edx-platform asset build, such that it may `search for node_modules outside of edx-platform <https://github.com/openedx/wg-developer-experience/issues/150> and `generate assets outside of edx-platform <https://github.com/openedx/wg-developer-experience/issues/151>`.
```

Comment:
I think your link syntax here is malformed, check the render you linked in the description.

Reply:
Yup, I'm missing the trailing underscore, thanks.